### DISCUSSION TITLE
Verification checks, eg missing media files?

### DISCUSSION THREAD

**Discussion by JonB451** [TheLastGimbus/GooglePhotosTakeoutHelper#362] [General]
Thanks loads for working on this. Could you please confirm whether your code does any verification checks between the Takeout and the Account itself, eg for any missing files that it failed to include? And do you know if this is because it excludes specific file types, eg HEIC, or some other metadata, eg about ownership? or is it just varied suboptimal performance, that often misses things but not always the same things?

**Comment by TheLastGimbus** [comment]
I take my **best** effort to make sure gpth doesn't loose people's photos, because that would be completely against whole idea of it - saving your time and headache

That's why I hardly do any PR's/small issues that popped up in recent months - because i don't have the time to *carefully* sit down and think them through whether they are safe

What gpths does now in these terms:
- finds all files that have video/image extension
  - 99.9% of extensions like jpg, heic, bmp etc match - once in a year someone comes up that their super-niche-adobe file format didn't match and i add it manually
- detects duplicates using files hash (their actual contents matching)
- moves everything to output except duplicates

What it does not do:
- check anything with google account
- check whether the user actually selected all their photos
- check if everything was copied, because... i know it was, lol... if the copy algorythm was broken, the "verify check" algorythm would be too

**Reply by JonB451** [reply]
ok, thanks for the answer, "No." Very much appreciated. And again, thanks for what you are aiming to do.
